core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -315,7 +314,7 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorAdded(executorId: String): Unit = synchronized {
     if (!executorIds.contains(executorId)) {
       executorIds.add(executorId)
-      executorIds.foreach(onExecutorIdle)
+      onExecutorIdle(executorId)
```

Comment:
Wait, but if we simply don't remove `executorId` from `removeExecutor` when we've reached the minimum, we will try to remove it again as soon as the next time `schedule` is called. This means we'll be constantly calling `removeExecutor` on that executor and failing. I think we should mark it specially as "should be removed as soon as new executors join" instead. (and we can do this with the -1 thing I mentioned earlier)